sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java

```diff
@@ -75,7 +76,7 @@ public DefaultServiceSpec(
                 ? DcosConstants.MESOS_MASTER_ZK_CONNECTION_STRING : zookeeperConnection;
         this.pods = pods;
         this.replacementFailurePolicy = replacementFailurePolicy;
-        this.user = user;
+        this.user = StringUtils.isEmpty(user) ? DcosConstants.DEFAULT_SERVICE_USER : user;
```

Comment:
Do we allow a username that contains only spaces? I checked with a quick unit test, and this failed. We could consider using `isBlank(user)` here too.